p2p/host/basic/natmgr.go

```diff
@@ -37,7 +37,7 @@ func NewNATManager(net network.Network) NATManager {
 //  * closing the natManager closes the nat and its mappings.
 type natManager struct {
 	net   network.Network
-	natmu sync.RWMutex
+	natMx sync.RWMutex
```

Comment:
Let's leave the name (`mu` stands for `mutex` and is a pretty common suffix in go).

Reply:
I've renamed it because in holepuncher we use "Mx".
https://github.com/libp2p/go-libp2p/blob/f383f4681f75b004ec2da0f4e9df0983bfbc65d2/p2p/protocol/holepunch/holepuncher.go#L36-L53

Reply:
Ah, I see. Eh, that's fine.